pkg/apis/datadoghq/v1alpha1/test/new.go

```diff
@@ -86,6 +87,7 @@ func NewDefaultedDatadogAgent(ns, name string, options *NewDatadogAgentOptions)
 			Log:                datadoghqv1alpha1.LogSpec{},
 			Process:            datadoghqv1alpha1.ProcessSpec{},
 		},
+		Site: options.Site,
```

Comment:
this causes a panic when `options` is `nil`
it should be moved to the block below when we check `if options != nil`

Reply:
🤦 Should have caught it earlier. Thank you!